test/upgrade/prober/wathola/config/tracing.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"go.uber.org/zap"
+	"knative.dev/pkg/tracing"
+	tracingconfig "knative.dev/pkg/tracing/config"
+)
+
+func SetupTracing() {
+	config, err := tracingconfig.JSONToTracingConfig(Instance.TracingConfig)
+	if err != nil {
+		Log.Warnf("Tracing configuration is invalid, using the no-op default", zap.Error(err))
+	}
+	if err = tracing.SetupStaticPublishing(Log, "", config); err != nil {
+		Log.Warnf("Error setting up trace publishing", zap.Error(err))
```

Comment:
I think it missing a return, if tracing config is invalid. Also, if tracing config is valid, but can't be set up it should fail fast.
```suggestion
		Log.Warnf("Tracing configuration is invalid, using the no-op default", zap.Error(err))
		return
	}
	if err = tracing.SetupStaticPublishing(Log, "", config); err != nil {
		Log.Fatalf("Error setting up trace publishing", zap.Error(err))
```

Reply:
It is not required because the function JSONToTracingConfig returns (in the event of error) a no-op configuration and it is used in the following function.
This patter is used for example [here](https://github.com/knative/eventing/blob/main/pkg/adapter/v2/config.go#L183) but it is on other places as well.

Reply:
Okay. I could agree about the return, but the second should fail instead of warn. Just as you showed the other example [here](https://github.com/knative/eventing/blob/main/pkg/adapter/v2/config.go#L183).

Reply:
OK. Fixing...